### PR TITLE
Gmail add history details to the pipeline

### PR DESCRIPTION
**Pull request by HazalCiplak**
https://github.com/elifesciences/data-hub-issues/issues/159
and 
https://github.com/elifesciences/data-hub-issues/issues/151

### CODE REVIEW

File: dags/gmail_data_import_pipeline.py [hunk 1]
```diff
@@ -168,6 +170,47 @@ def gmail_link_message_thread_ids_etl(**kwargs):
...
+    project_name = data_config.project_name
+
+    start_id = get_max_history_id_from_bq(
+                            project_name=data_config.project_name,
+                            dataset=dataset_name,
+                            column_name='historyId',
+                            table_name=data_config.table_name_thread_details
+                        )
+
+    with TemporaryDirectory() as tmp_dir:
```
Comment (de-code) [inline dags/gmail_data_import_pipeline.py:187]: This block now appears four times (and passes the [rule of three](https://blog.codinghorror.com/rule-of-three/)). It seems to be worth refactoring or were you planning to do that as a separate PR. All that seems to be different is the DataFrame and the target BigQuery table. In `data-science-dags` I chose to follow Pandas' `to_gbq` interface. But could be any interface.

File: dags/gmail_data_import_pipeline.py [hunk 2]
```diff
@@ -249,5 +299,12 @@ def gmail_thread_details_etl(**kwargs):
 
 # pylint: disable=pointless-statement
 # define dependencies between tasks in the DAG
-get_data_config_task >> [gmail_label_data_etl_task, gmail_link_message_thread_ids_etl_task]
-gmail_link_message_thread_ids_etl_task >> gmail_thread_details_etl_task
+get_data_config_task >> [
+                            gmail_label_data_etl_task,
```
Comment (de-code) [inline dags/gmail_data_import_pipeline.py]: minor: I would personally reduce the indentation and put a blank line between after `]` to make it easier to see that it is a separate statement.

File: dags/gmail_data_import_pipeline.py [hunk 3]
```diff
@@ -168,6 +170,47 @@ def gmail_link_message_thread_ids_etl(**kwargs):
...
 
+def gmail_history_details_etl(**kwargs):
+    data_config = data_config_from_xcom(kwargs)
+    user_id = get_gmail_user_id()
+    table_name = data_config.table_name_history_details
+    dataset_name = data_config.dataset
+    project_name = data_config.project_name
+
+    start_id = get_max_history_id_from_bq(
+                            project_name=data_config.project_name,
```
Comment (de-code) [inline dags/gmail_data_import_pipeline.py]: minor: perhaps reduce indentation?

**Review by de-code — APPROVED** [review]

**Comment by de-code** [comment]
And the usual nudge to add automated tests :-)

File: dags/gmail_data_import_pipeline.py (hunk 1 above)
Comment (HazalCiplak) [reply dags/gmail_data_import_pipeline.py:187]: have created a new issue

Comment (de-code) [reply dags/gmail_data_import_pipeline.py:187]: Just adding the link to the issue: https://github.com/elifesciences/data-hub-issues/issues/161
